pyccel/ast/builtins.py

```diff
@@ -431,13 +430,11 @@ def __init__(self, *args, **kwargs):
                 raise TypeError('cannot determine the type of {}'.format(self))
 
             shapes = [a.shape for a in args]
-
+            self._rank = max(a.rank for a in args) + 1
             if all(sh is not None for sh in shapes):
-                assert all(sh==shapes[0] for sh in shapes)
                 self._shape = (LiteralInteger(len(args)), ) + shapes[0]
```

Comment:
Does it still make sense to save the shape in the other dimensions if it is not the same everywhere? (I am assuming that is why the assertion was removed)

Reply:
we can not verify if they are the same, we just assume that they are equal for the moment, and if there is a problem it will be a run time error